src/commands/shell.mjs

```diff
@@ -105,11 +109,30 @@ async function shellCommand(argv) {
         shell.prompt();
       },
     },
+    {
+      cmd: "togglePerformanceHints",
+      help: "Enable or disable performance hints. Disabled by default. If enabled, outputs performance hints for the most recent query.",
+      action: () => {
+        shell.context.performanceHints = !shell.context.performanceHints;
+        logger.stderr(
+          `Performance hints in shell: ${shell.context.performanceHints ? "on" : "off"}`,
+        );
+        shell.prompt();
+      },
+    },
   ].forEach(({ cmd, ...cmdOptions }) => shell.defineCommand(cmd, cmdOptions));
 
   return completionPromise;
 }
 
+const getArgvOrCtx = (key, argv, ctx) => {
```

Comment:
I think we could get away with mutating argv for these shell commands to tweak settings, but I haven't gone too deep into what weird side effects that may create. I started looking into a `.setOption` command instead of a list of specific `.toggleX` commands, and was thinking that mutating argv might be necessary.